Patch-Source: https://github.com/gdm-settings/gdm-settings/pull/277
---
From 6241a88446dd79701dda7874fb3df7c934fe8e0f Mon Sep 17 00:00:00 2001
From: fossdd <fossdd@pwned.life>
Date: Wed, 4 Dec 2024 21:57:35 +0100
Subject: [PATCH] Avoid using machinectl, use su instead

machinectl is unavailable on non-systemd-based systems like Alpine Linux
or Chimera Linux. su is a alternative that works on all Unix systems.
---
 gdms/settings.py | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/gdms/settings.py b/gdms/settings.py
index 05eee976..e7ca46e8 100644
--- a/gdms/settings.py
+++ b/gdms/settings.py
@@ -630,10 +630,9 @@ def apply_user_display_settings() -> bool:
     shutil.copyfile(user_monitors_xml, temp_monitors_xml)
     os.chmod(temp_monitors_xml, 0o644)
 
-    _commands.add(['machinectl', 'shell', f'{gresource.GdmUsername}@', '/usr/bin/env',
-                     'gsettings', 'set', 'org.gnome.mutter', 'experimental-features',
-                     '"[\'scale-monitor-framebuffer\']"',
-                     '&>/dev/null',
+    _commands.add(['su', f'{gresource.GdmUsername}',
+                     '-s', '/bin/sh',
+                     '-c', '"dbus-run-session gsettings set org.gnome.mutter experimental-features \\"[\'scale-monitor-framebuffer\']\\""'
                    ])
 
     _commands.add(['install', '-Dm644',
